sql/src/main/java/io/crate/jobs/PageDownstreamContext.java

```diff
@@ -114,7 +114,7 @@ public void setBucket(int bucketIdx, Bucket rows, boolean isLast, PageResultList
                     mergeBucketsAndEmit();
                 } else {
                     rowReceiver.fail(lastThrowable);
-                    close(lastThrowable);
+                    finishDownstream(lastThrowable);
```

Comment:
We renamed this to `releaseListenersAndCloseContext` on the batch-iterator branch, maybe you want to do this here as well - because the name implies that it also calls `rowReceiver.fail/finish` which it doesn't.